hbase_sql/src/main/java/com/splicemachine/mrio/api/core/AbstractSMInputFormat.java

```diff
@@ -191,6 +192,39 @@
         }
     }
 
+    /**
+     * Checks the sequence of split rows. If any gap is found, the splitting has to be recalculated again.
+     *
+     * @param lss
+     * @return
+     */
+    protected boolean isRefreshNeeded(List<InputSplit> lss) {
```

Comment:
Can you add a check for the scan start/stop keys? Making sure there's no gap between the scan start key and the first split, or between the last split and the scan stop key. This check should also be made when there's only one split.

Reply:
More checks are added.